File: src/com/easy/SingleNumber_136.java
/**
 * 
 */
package com.easy;

/**
 * @author Chris Lin
 * @version 1.0
 */
public class SingleNumber_136 {
    
    // Runtime: 0 ms, faster than 100.00%
    // Memory Usage: 41.7 MB, less than 20.36%
    public int singleNumber(int[] nums) {
        int mResult = 0;
        for (int num : nums) {
            mResult ^= num;
        }

        return mResult;
    }
}

File: Cpp_src/easy/single_number_136.cc
#include <iostream>
#include <vector>
using namespace std;

// Use bit manipulation
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 16 ms, faster than 67.73%
// Memory Usage: 9.5 MB, less than 100.00%
class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        int result = 0;
        for (int num : nums)
        {
            result ^= num;
        }
        return result;
    }
};